### ISSUE TITLE
Possible EventEmitter memory leak detected

### ISSUE THREAD

**Issue by sbat** [appfeel/node-pushnotifications#44]
I get this warning:

> (node:70543) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 wakeup listeners added. Use emitter.setMaxListeners() to increase limit

The warning comes from the APN Module:  https://github.com/node-apn/node-apn/issues/518

See someone else this error? Have someone a solution?

**Comment by alex-friedl** [comment]
Hi @sbat,
from the issue you referenced it looks like it's an open issue with node-apn?

**Comment by DerTieran** [comment]
I think this happens because you always create a [new connection](https://github.com/appfeel/node-pushnotifications/blob/master/src/sendAPN.js#L33) and never close it.
```js
const connection = new apn.Provider(settings.apn);
return connection.send(message, regIds)
  .then((response) => {
    /* ... */
    connection.shutdown();
    return resumed;
  })
  .catch((error) => {
    connection.shutdown();
    throw error;
  })
```
But as mentioned in #47 it would be better to reuse the connection and offer a `push.shutdown()` or something like this.

**Comment by heyjared** [comment]
Is there a way to manually shutdown the connection to the provider?

**Comment by flipace** [comment]
When will #55 be merged and released?

**Comment by jpike88** [comment]
Looks like the pull request doesn't really fix the issue apparently. Do yourself a favour everyone and update to Node 8.8.1+... I have released a package that eliminates the memory leak issue.

https://github.com/streaka/node-pushnotifications-http2

**Comment by alex-friedl** [comment]
Merging #55  as a quick fix for the memory leak issue. 
Will look into changing the implementation to only use one Provider instance, if feasible.

**Comment by alex-friedl** [comment]
Fixed after merge of #55
